packages/mattermost-redux/src/types/posts.ts

```diff
@@ -145,3 +147,10 @@ export declare type OpenGraphMetadata = {
     url?: string;
     images: OpenGraphMetadataImage[];
 };
+
+export declare type PostPreviewMetadata = {
+    id: string;
+    user_id: string;
+    channel_display_name: string;
+    team_name: string;
```

Comment:
These names might also be considered confidential as well like, for example, if someone posted a link to a customer channel in a public channel on community.

Reply:
None of the post metadata appears if the user doesn't have permission to view the post.

Reply:
Oh that's good to hear. I wasn't sure how that was handled since I haven't looked at the server code at all.